test/integration/device-api/v1.spec.ts

```diff
@@ -85,4 +89,87 @@ describe('device-api/v1', () => {
 				});
 		});
 	});
+
+	describe('POST /v1/restart', () => {
+		// Actions are tested elsewhere so we can stub the dependency here
+		let doRestartStub: SinonStub;
+		beforeEach(() => {
+			doRestartStub = stub(actions, 'doRestart').resolves();
+		});
+		afterEach(async () => {
+			doRestartStub.restore();
+			// Remove all scoped API keys between tests
+			await db.models('apiSecret').whereNot({ appId: 0 }).del();
+		});
+
+		it('validates data from request body', async () => {
```

Comment:
Why test these methods here again? Why not test them with the full test for managing the app lifecycle? I understand maybe it is not practical for all conditions (like simulating locks or edge error cases), but I would remove any redundant code as it is just a maintenance burden.

Reply:
This is a step towards my goal, where I would like to add a PR to add an input validation middleware, after which all input validation tests can be deduplicated and tested in that middleware. In the meantime, I can remove it all if the degree of duplication is not satisfactory?

Reply:
You can leave the validation tests since you have a plan to remove them. The duplication is fairly minor as well as the lifecycle tests do not test all input scenarios.